docs/zkapps/faq.mdx

```diff
@@ -95,3 +95,66 @@ This example fails for two reasons:
 
 1. `n.toString()` can't be used in circuit code at all. It throws an error during `SmartContract.compile()` because during `compile()`, variables like `n` don't have any JS values attached to them; they represent abstract variables used to build up an abstract arithmetic circuit. So, in general, you can't use any of the methods that read out the JS value of your Field elements: `Field.toString()`, `Field.toBigInt()`, `Bool.toBoolean()` etc.
 2. More subtly, your methods must create the same constraints every time because a proof cannot be verified against a verification key for a differing set of constraints. The code above adds `x.equals(y).assertFalse()` _on condition of_ the value of `n` which leads to constraints varying between executions of the proof.
+
+### Why is the variable currentState set to a value retrieved from the blockchain and then immediately compared to that value?
+
+As represented in the tutorial example code:
+
+```TypeScript
+const currentState = this.num.get();
+
+this.num.assertEquals(currentState);
+```
+
+- The first line of code executes before the proof is generated. 
+- The second line of code creates a precondition that is checked when the proof is sent to the blockchain to be verified. 
```

Comment:
The second line of code creates a precondition that is checked when the proof is sent in a transaction the blockchain to be verified.